File: backend/app/schemas/memory.py
from pydantic import BaseModel


class DrawMemoryResponse(BaseModel):
    stone_id: str
    stone_type: str
    stone_color: str
    content: str
    image_url: str | None
    created_at: str
    ai_message: str
    hp_change: int
    hp_after: int

File: frontend/lib/api/memory.ts
import { apiRequest, type ApiResponse } from "@/lib/api/client";

export interface DrawMemoryData {
  stoneId: string;
  stoneType: string;
  stoneColor: string;
  content: string;
  imageUrl: string | null;
  createdAt: string;
  aiMessage: string;
  hpChange: number;
  hpAfter: number;
}

export async function drawMemory(): Promise<ApiResponse<DrawMemoryData>> {
  return apiRequest<DrawMemoryData>("/memory/draw", {
    method: "POST",
  });
}
